### ISSUE TITLE
Unable to run against Azure Stack

### ISSUE THREAD

**Issue by charliejllewellyn** [Azure/azure-sdk-for-python#1660]
I am indirectly using this module via Ansible against Azure Stack. I have run into an issue that prevents Ansible from running because the same endpoints as public Azure are not available.

When specifying a metadata discovery endpoint (export AZURE_CLOUD_ENVIRONMENT="https://management.local.azurestack.external/metadata/endpoints?api-version=2016-01-01") I receive the following error.

```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py", line 259, in <module>
    main()
  File "/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py", line 256, in main
    AzureRMResourceGroup()
  File "/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py", line 140, in __init__
    supports_tags=True)
  File "/tmp/ansible_r4mrl9/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py", line 276, in __init__
  File "/usr/lib/python2.7/site-packages/msrestazure/azure_active_directory.py", line 359, in __init__
    self._configure(**kwargs)
  File "/usr/lib/python2.7/site-packages/msrestazure/azure_active_directory.py", line 152, in _configure
    resource = self.cloud_environment.endpoints.management
  File "/usr/lib/python2.7/site-packages/msrestazure/azure_cloud.py", line 84, in __getattribute__
    "is not set but is used.".format(name))
msrestazure.azure_cloud.CloudEndpointNotSetException: The endpoint 'management' for this cloud is not set but is used.

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py\", line 259, in <module>\n    main()\n  File \"/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py\", line 256, in main\n    AzureRMResourceGroup()\n  File \"/tmp/ansible_r4mrl9/ansible_module_azure_rm_resourcegroup.py\", line 140, in __init__\n    supports_tags=True)\n  File \"/tmp/ansible_r4mrl9/ansible_modlib.zip/ansible/module_utils/azure_rm_common.py\", line 276, in __init__\n  File \"/usr/lib/python2.7/site-packages/msrestazure/azure_active_directory.py\", line 359, in __init__\n    self._configure(**kwargs)\n  File \"/usr/lib/python2.7/site-packages/msrestazure/azure_active_directory.py\", line 152, in _configure\n    resource = self.cloud_environment.endpoints.management\n  File \"/usr/lib/python2.7/site-packages/msrestazure/azure_cloud.py\", line 84, in __getattribute__\n    \"is not set but is used.\".format(name))\nmsrestazure.azure_cloud.CloudEndpointNotSetException: The endpoint 'management' for this cloud is not set but is used.\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}
```

We have worked around this by setting both the "management" and "resource_manager" on _populate_from_metadata_endpoint in msrestazure/azure_cloud.py. Can you confirm if this is an error with this module not supporting Azure Stack or an error with Azure Stack not creating the correct endpoints?

**Comment by lmazuel** [comment]
Could you download this URL using `curl` or `requests` and post the JSON you received (remove all personal information, the only thinkg I need is the JSON structure).
Then, I will check if it's a AS bug.

In the meantime, could you change `"/usr/lib/python2.7/site-packages/msrestazure/azure_active_directory.py", line 152` to call the `resource_manager` attribute instead of `management` ?

@derekbekoe FYI, probably the CLI will got the same issue.

**Comment by lmazuel** [comment]
Fixed in msrestazure 0.4.18
